### PR TITLE
Update README

### LINKED ISSUE #59: Incorrect Git URL & missing `cd` in README
README has incorrect Git URL and lacks `cd` after cloning.

```diff
-git clone https://ntut-open-source-club/practical-tools-for-simple-design.git
+git clone https://github.com/ntut-open-source-club/practical-tools-for-simple-design.git
+cd practical-tools-for-simple-design

 cmake -B build
 cmake --build build
```

### LINKED ISSUE #78: Add extra lib needed to README
> The configuration on Ubuntu may have some issue
I realize the step may be just clone it and configure the project with CMake (with clang and clion also). But I still need some packages to make it configure successfully. It could be mentioned on documentation or somewhere 

```
sudo apt-get install libgl1-mesa-dev libxext-dev
```

@ntut-xuan 

[source](https://discord.com/channels/1093814266630512681/1093814266630512684/1194683458467921970)

### PR DESCRIPTION
**Pull request by ntut-rick**
- #67

  - [x] fix #59 
  - [x] fix #78 
  - [x] add reference Good reference from [OOP2023f Environment Setup](https://hackmd.io/@OOP2023f/rk2-8cVCh) (by @NOOBDY and @ntut-xuan )
  - [x] Add README in doxygen docs

  
TODO

- [ ] different between LGF and PTSD
  - Extra info from [the slide](https://hackmd.io/CtUemwatQCGe9pgvAGsGJg?both#/) (available as of Dec 22, by @NOOBDY).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,35 @@
...
+|                                | Windows | macOS | Linux |
+|:------------------------------:|:-------:|:-----:|:-----:|
+|             CLion              |    V    |   V   |   V   |
+|        VSCode[^codeoss]        |    V    |   V   |   V   |
+| Visual Studio[^novs][^vsmacos] |    ?    |   X   |   X   |
+|             No IDE             |    V    |   V   |   V   |
+
+[^vsmacos]: [Microsoft Kills Visual Studio for Mac](https://visualstudiomagazine.com/articles/2023/08/30/vs-for-mac-retirement.aspx)
+[^codeoss]: On Linux the support of Code - OSS and VSCodium aren't guaranteed.
+[^novs]: We don't provide support for Visual Studio, but building PTSD on VS is available.
```
Comment (NOOBDY) [inline README.md:14]: I feel like we could provide partial support, but due to lack of testing there may or may not be more issues on VS

**Review by NOOBDY — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (ntut-rick) [reply README.md:14]:
```suggestion
[^novs]: We could provide partial support, but due to lack of testing there may or may not be more issues on VS. anyway, building PTSD on VS is available.
```

**Review by NOOBDY — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -1,16 +1,35 @@
 # Practical Tools for Simple Design
 
+###### Officially Supported Platforms and IDEs
+
+|                                | Windows | macOS | Linux |
+|:------------------------------:|:-------:|:-----:|:-----:|
+|             CLion              |    V    |   V   |   V   |
+|        VSCode[^codeoss]        |    V    |   V   |   V   |
+| Visual Studio[^novs][^vsmacos] |    ?    |   X   |   X   |
+|             No IDE             |    V    |   V   |   V   |
```
Comment (HEKEPOIU) [inline README.md:10]: WTF? GARBAGE

File: README.md [hunk 3]
```diff
@@ -1,16 +1,35 @@
...
+
+[^vsmacos]: [Microsoft Kills Visual Studio for Mac](https://visualstudiomagazine.com/articles/2023/08/30/vs-for-mac-retirement.aspx)
```
Comment (HEKEPOIU) [inline README.md:12]: footnote not good, please put closely.

File: README.md (hunk 2 above)
Comment (ntut-rick) [reply README.md:10]: Some framework I know doesn't officially provide technical support on no IDE develop.

File: README.md (hunk 3 above)
Comment (ntut-rick) [reply README.md:12]: Hmm

**Pull request merged**